packages/react/src/useFetch.ts

```diff
@@ -190,11 +206,17 @@ export function useFetch<T = string>(path: string, options?: FetchHookOptions):
                 }
               }
             }
-          })().catch((error) => {
-            if (!abortContoller.signal.aborted) {
-              dispatchError(mounted, dispatch, error, response);
-            }
-          });
+
+            mergedOptions.onStreamComplete?.();
```

Comment:
Another approach here is for the user to do this:

```js
const previousStreaming = useRef(streaming);

useEffect(() => {
  if(previousStreaming.current && !streaming) {
    // on stream complete
  }
  previousStreaming.current = streaming;
}, [streaming])
```

Which would avoid your callback issue, but kind of annoying for users to have to write. I guess it doesn't deal with aborts either, especially the ones that would be ignored by placing it here inside of the `try`

(scrap the idea, but leaving the comment here in case it sparks anything)